posts/how-to-create-a-quick-react-and-angular-app-with-jhipster-lite.md

```diff
@@ -0,0 +1,457 @@
+
+```javascript
+import axios from 'axios';
+ 
+import { getLocalStorage } from '@/common/services/storage';
+import { BeersFunctionType } from '@/beers/interface';
+ 
+const BEERS_URL = '/api/beers';
+ 
+export const getBeers = ({ setBeers }: BeersFunctionType) => {
+ const headers = {
+   'Authorization': 'Bearer ' + getLocalStorage('token')
+ }
+  return axios
+   .get(BEERS_URL, {
+     headers: headers
+   })
+   .then(response => {
+     setBeers(response.data.beers);
+   })
+   .catch(error => {
+     console.error(error);
+   });
+};
+```
+
+In the authentication code in the generated project, the JWT is stored in local storage during login. We then place this token in the header, in order to make the service call. The response from the service is returned in an array of beer objects.
+
+## Display the results of a Beers service call
+
+Create a `beers.tsx` file to output the results
+
+```javascript
+import { useState, useEffect } from 'react';
+import { getBeers } from '@/beers/services/beers';
+import { IBeer } from '@/beers/interface';
+ 
+export default function Beers() {
+   const [beers, setBeers] = useState<Array<IBeer>>([]);
+ 
+   useEffect(() => {getBeers({setBeers})}, []);
+ 
+   return (
+       <main style={{ padding: "1rem 0" }}>
+           <h2>Beers</h2>
+           <ul>
+               {beers.length > 0 && beers.map(beer => (<li>{beer.name}: ${beer.unitPrice}</li>))}
+           </ul>
+       </main>
+   );
+}
+```
+
+The code here makes the call to the `getBeers` function and displays the beer names and prices in an unordered list.
+
+## Add a router and a route to the new beers page
+
+Run `npm install react-router-dom` to install a router
+
+To add the route, modify the `src/main/webapp/app/index.tsx` file like the following
+
+```typescript
+import React from 'react';
+import { createRoot } from 'react-dom/client';
+import './index.css';
+import App from '@/common/primary/app/App';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
+import Beers from "./beers/beers"
+ 
+const container = document.getElementById('root');
+const root = createRoot(container!);
+root.render(
+ <React.StrictMode>
+   <BrowserRouter>
+     <Routes>
+       <Route path="/" element={<App />} />
+       <Route path="/beers" element={<Beers />} />
+     </Routes>
+   </BrowserRouter>
+ </React.StrictMode>
+);
+```
+
+The imports are new and now we have a route for the home page and the new beers page.
+
+## Modify the home page to add a link to the new beers page
+
+The home page is located at `src/main/webapp/app/common/primary/app/App.tsx`
+
+Add a Link import for the react router
+
+```typescript
+import { Link } from "react-router-dom";
+```
+
+Add a link to the new beers page
+
+```html
+<Link to="/beers">I love beer</Link>
+```
+
+## Add some beers with the service for display
+
+Restart your application to make sure you have the latest changes
```

Comment:
I would assume there is a live reload option to detect changes, no?

Reply:
I decided the blog was long enough so I did not want to get into that. You have to run the npm serve and hit the other port to get the live reload.

Reply:
Ah okay, yeah I assume most FE people would understand. From me reading I'm like, "oh if I have to do that for every code change it is kind of a bad developer experience"